compendium/DeclarativeServices/src/manager/ComponentConfigurationImpl.cpp

```diff
@@ -234,7 +234,10 @@ namespace cppmicroservices
                         {
                             for (auto const& config : configs)
                             {
-                                configNotifier->CreateFactoryComponent(config->GetPid(), mgr);
+                                auto componentFactory = configNotifier->GetComponentFactory();
+                                componentFactory->CreateFactoryComponent(config->GetPid(),
```

Comment:
Its not clear to me how exceptions and failure are supposed to be handled. can you clarify what is supposed to happen in the case of an exception being thrown? e.g. are all `config` in `configs` supposed to have a factory component created if one or more cause an exception?

Reply:
Any exceptions thrown in ComponentConfigurationImpl::Initialize are caught and logged by CMEnabledState::CreateConfigurations. This is true for factory components but it's also true for normal components that have been created with a manifest.json file rather than virtual metadata. OSGI has a bunch of places where it talks about errors happening and the result is that the error should be logged.